doc/api/url.md

```diff
@@ -2,136 +2,221 @@
+URL string.
+
+The formatting process essentially operates as follows:
+
+* A new empty string `result` is created.
+* If `urlObj.protocol` is a string, it is appended as-is to `result`.
+* Otherwise, if `urlObj.protocol` is not `undefined` and is not a string, an
+  [`Error`][] is thrown.
+* For all string values of `urlObj.protocol` that *do not end* with an ASCII
+  colon (`:`) character, the literal string `:` will be appended to `result`.
+* If either the `urlObj.slashes` property is true, `urlObj.protocol` begins
+  with one of `http`, `https`, `ftp`, `gopher`, or `file`, or `urlObj.protocol` 
+  is `undefined`, the literal string `//` will be appended to `result`.
+* If the value of the `urlObj.auth` property is `truthy`, and either
+  `urlObj.host` or `urlObj.hostname` are not `undefined`, the value of
+  `urlObj.auth` will be coerced into a string and appended to `result` followed
+   by the literal string `@`.
+* If the `urlObj.host` property is `undefined` then:
+  * If the `urlObj.hostname` is a string, it is appended to `result`.
+  * Otherwise, if `urlObj.hostname` is not `undefined` and is not a string, an
+    [`Error`][] is thrown.
+  * If the `urlObj.port` property value is `truthy`, and `urlObj.hostname` is
+    not `undefined`:
+    * The literal string `:` is appended to `result`, and
+    * The value of `urlObj.port` is coerced to a string and appended to
+      `result`.
+* Otherwise, if the `urlObj.host` property value is `truthy`, the value of
+  `urlObj.host` is coerced to a string an appended to `result`.
+* If the `urlObj.pathname` property is a string that is not an empty string:
+  * If the `urlObj.pathname` *does not start* with an ASCII forward slash (`/`),
+    then the literal string '/' is appended to `result`.
+  * The value of `urlObj.pathname` is appended to `result`.
+* Otherwise, if `urlObj.pathname` is not `undefined` and is not a string, an
+  [`Error`][] is thrown.
+* If the `urlObj.search` property is `undefined` and if the `urlObj.query`
+  property is an `Object`, the literal string `?` will be appended to `result`
+  followed by the output of calling the `querystring` module's `stringify()`
+  function passing the value of `urlObj.query`.
+* Otherwise, if `urlObj.search` is a string:
+  * If the value of `urlObj.search` *does not start* with the ASCII question
+    mark (`?`) character, the literal string `?` is appended to `result`.
+  * The value of `urlObj.search` is appended to `result`.
+* Otherwise, if `urlObj.search` is not `undefined` and is not a string, an
+  [`Error`][] is thrown.
+* If the `urlObj.hash` property is a string:
+  * If the value of `urlObj.hash` *does not start* with the ASCII hash (`#`)
+    character, the literal string `#` is appended to `result`.
+  * The value of `urlObj.hash` is appended to `result`.
+* Otherwise, if the `urlObj.hash` property is not `undefined` and is not a
+  string, an [`Error`][] is thrown.
+* `result` is returned.
+
```

Comment:
I don't like the change here. At least in GH HTML this looks very confusing due to the first part in camel case and it not being aligned with the left edge of the page. Semantically it got better though. I don't have a better idea though and final rendering would be interesting